Applications/Spire/Source/Charting/ChartView.cpp

```diff
@@ -216,35 +217,49 @@ void ChartView::reset_crosshair() {
   m_crosshair_pos.reset();
 }
 
-std::tuple<ChartPoint, ChartPoint> ChartView::get_region() const {
-  return {m_top_left, m_bottom_right};
+ChartView::Region ChartView::get_region() const {
+  return m_region;
 }
 
 void ChartView::set_region(const ChartPoint& top_left,
     const ChartPoint& bottom_right) {
-  if(top_left == m_top_left && bottom_right == m_bottom_right) {
+  if(top_left == m_region.m_top_left &&
```

Comment:
Define ```operator ==``` and ```operator !=``` on ```Region```.

Reply:
Should these be defined publicly or in the anonymous namespace?

Reply:
Nevermind, it's a struct.

Reply:
Question still stands; I forgot about the problem with global vs. member overloaded operators.

Reply:
The operators should be methods added to ```Region```.